ibis/common/grounds.py

```diff
@@ -69,17 +69,28 @@ def validate(self, this, arg):
             return self.validator(arg, this=this)
 
 
-@functools.singledispatch
-def _get_hashable_object(value: Any) -> Hashable:
-    return value
+class Immutable(Hashable):
```

Comment:
`behemoth2` already has this change

Reply:
I'd like to merge this PR before we merge behemoth2.

Reply:
Sure.